drivers/gpu/drm/meson/meson_cache_control.c

```diff
@@ -69,53 +70,79 @@ static void meson_drm_ump_osk_msync(struct drm_gem_cma_object *cma_obj, void *vi
 		}
 	}
 
-	if (!cma_obj)
+	if (!gem_obj)
 		return;
-	gem_obj = &cma_obj->base;
 
 	DBG_MSG(3, ("meson_drm_ump_osk_msync(): Flushing CPU L2 Cache\n"));
 
-	/* Flush L2 using physical addresses
-	 * Our allocations are always contiguous (GEM CMA), so we have only one mem block */
-
-	if (offset >= gem_obj->size) {
-		offset -= gem_obj->size;
-		return;
-	}
-
-	if (offset) {
-		start_p = (u32)cma_obj->paddr + offset;
-		/* We'll zero the offset later, after using it to calculate end_p. */
+	if (gem_obj->is_scattered) {
+		struct meson_drm_gem_scattered_object *meson_gem = to_meson_drm_gem_scattered_obj(gem_obj);
+		struct scatterlist *sgl;
+		int i;
+
+		for_each_sg(meson_gem->sgt->sgl, sgl, meson_gem->sgt->nents, i) {
+			// start_p = (u32)__pa(page_address(sg_page(sgl)));
```

Comment:
nit: Use C-style /* */ comments

You should also probably have a comment saying why you're not using this if you left it commented out for some reason.